libexec/bats-exec-test

```diff
@@ -331,7 +331,20 @@ BATS_OUT="${BATS_TMPNAME}.out"
 
 bats_preprocess_source() {
   BATS_TEST_SOURCE="${BATS_TMPNAME}.src"
-  . bats-preprocess <<< "$(< "$BATS_TEST_FILENAME")"$'\n' > "$BATS_TEST_SOURCE"
+  BATS_TEST_SOURCE_WIP="${BATS_TMPNAME}.wip"
+  { tr -d '\r' < "$BATS_TEST_FILENAME"; echo; } | bats-preprocess > "$BATS_TEST_SOURCE_WIP"
```

Comment:
Why is this block of changes necessary? Also, there are a lot of processes getting spawned here; please rewrite them using Bash-only constructs. (See #8 for the rationale on eliminating every subshell possible. TL;DR: This can have a significant impact on performance, especially on Windows.)

Reply:
Given a bats script, this code will go through the file and if there is a `@test` or `include` statement it will call bats-preprocess. After the first round, it will scan the generated file again for `@test` or `include` statements. This is done until no such statements are encountered in the file. This is needed as an included bats script can have `@test` or `include` commands of its own, allowing for multiple levels of inclusion.  

There are indeed many processes and files involved. 
- As we cannot read and write to the same file at the same time, we need at minimum two files; one to read and one to write to. I am using the $BATS_TEST_SOURCE and the $BATS_TEST_SOURCE_WIP files as the source and destination files, alternating their role in each loop round. 
- I could combine the two greps into one. To avoid the  redirection, I could also make grep silent with `-q` but that would mess up the error exit codes. Redirecting to `/dev/null/` [should be fast anyway](https://stackoverflow.com/questions/2725144/can-i-get-a-faster-output-pipe-than-dev-null). Is it possible to grep the file using only bash constructs?
- The `mv` is the fastest option I could think of from the different alternatives. It doesn't open a file and it basically renames. 
- Another note, line 342 should be changed according to your commit cf9a3b8a.

Compared to bats on the master branch, this will add one call to grep and one to mv when processing a file without an `include` statement.

I cannot think of any other optimization, other than using a different approach altogether. Do you think there could be another approach to implementing this? Maybe loading the initial file contents in a variable and manipulating it instead of using files?